### ISSUE TITLE
Cannot list certificate which was just created

### ISSUE THREAD

**Issue by lilithyang2018** [TykTechnologies/tyk#4776]
**Branch/Environment/Version**
Tyk Gateway v3.0.12
redis:6.0.12-alpine

**Describe the bug**
Cannot list certificate which was just created

**Reproduction steps**
1. flush your redis db
2. Call List Certificates API:   GET /tyk/certs
3. Call Add certificate API: POST /tyk/certs
4. Call List Certificates API again

**Actual behavior**
You will see a "{certs: null}" in step 4's response body.

**Expected behavior**
You should see the id of the cert you just uploaded in step 3

P.S. if we remove step 2(don't call list before add), we can get the expected.

**Screenshots/Video**
![cert saved](https://user-images.githubusercontent.com/39897706/218982991-c4f9a52c-e7e8-43a9-8131-3c8789343a5d.png)
![cannot see in list](https://user-images.githubusercontent.com/39897706/218983065-ee726eb1-9f0b-404b-a4ac-58b2b43f2886.png)

**Logs (debug mode or log file):**
![tyk log](https://user-images.githubusercontent.com/39897706/218983301-f58cc351-7fb3-40dd-86b0-5043e1aff3fa.png)

**Comment by oluwaseyeayinla** [comment]
Reproduced the issued. Thanks for spotting the issue.

**Comment by SDkie** [comment]
Hey @oluwaseyeayinla 

I am always getting null certs, even after adding certificate. 
```
{
    "certs": null
}
```

Looks like [this](https://github.com/TykTechnologies/tyk/blob/master/certs/manager.go#L518) code is executed, because of which list is always empty. Not sure if this has anything to do with my deployment.

**Comment by SDkie** [comment]
@oluwaseyeayinla 

Also, add and list certificate on the tyk-dashboard is not calling tyk-gateway APIs? 
I am not facing this issue on the tyk-dashboard, but getting it when I call gateway-apis using Postman. 

<img width="1466" alt="Screenshot 2023-03-12 at 1 18 10 PM" src="https://user-images.githubusercontent.com/1321321/224531445-3b211332-77ff-4cef-81c4-2ebc414ae8ab.png">

**Comment by SDkie** [comment]
Hi @oluwaseyeayinla

I found the issue. I was not passing `org_id` to the API. Looks like with empty `org_id`, the API is not working properly.

**Comment by SDkie** [comment]
Hi @lilithyang2018 

Did you pass `org_id` in your test? I am not facing any issue if I pass `org_id`.

**Comment by lilithyang2018** [comment]
> Hi @lilithyang2018
> 
> Did you pass `org_id` in your test? I am not facing any issue if I pass `org_id`.

no,  it works well without the org_id in an older version(3.0.5),  but  have issue in the new version. **if the org_id is mandatory, i should receive the 400 response when create the cert**,  what do you think ?

**Comment by SDkie** [comment]
@lilithyang2018: Yes, you are correct if org_id is mandatory then API should give an error.

**Comment by oluwaseyeayinla** [comment]
@lilithyang2018 is correct. The gateway can be a standalone deployment and should work without passing an `org_id` since org_ids are a dashboard concept.

@SDkie Thanks for spotting that `org_id` make it work most of the time. I think I found where the issue is. There is a `-migrated` postfix that causes the problem when the org_id is empty. If I am not wrong, the `-migrated` postfix allows the system to detect if the org id cert list is empty. From the codebase, it looks like it is always set back to empty if an org id cert list isn't found. This would cause problems for an empty org_id since there shouldn't be a cert list.

A workaround would be to delete the key `cert--index-migrated` from Redis directly. This only makes it work once and you would have to always delete the key every time. I'll drop my findings in the internal ticket TT-8211
